R/read_resource.R

```diff
@@ -348,6 +338,7 @@ read_resource <- function(package, resource_name) {
         ),
         col_names = col_names,
         col_types = col_types,
+        col_select = {{col_select}},
```

Comment:
It's a metaprogramming pattern in rlang, called embracing, we need this to inject the expression to `readr::read_delim()`. You'll notice that the tests will fail if you replace the `{{` with `{`. Have a look at this page for documentation: https://rlang.r-lib.org/reference/topic-data-mask.html


See also: 

---

https://rlang.r-lib.org/reference/topic-inject.html#embracing-with-

The embracing operator [{{](https://rlang.r-lib.org/reference/embrace-operator.html) is made specially for function arguments. It [defuses](https://rlang.r-lib.org/reference/topic-defuse.html) the expression supplied as argument and immediately injects it in place. The injected argument is then evaluated in another context such as a [data mask](https://rlang.r-lib.org/reference/topic-data-mask.html).

``` r
# Inject function arguments that might contain
# data-variables by embracing them with {{ }}
mean_by <- function(data, by, var) {
  data %>%
    dplyr::group_by({{ by }}) %>%
    dplyr::summarise(avg = mean({{ var }}, na.rm = TRUE))
}

# The data-variables `cyl` and `disp` inside the
# env-variables `by` and `var` are injected inside `group_by()`
# and `summarise()`
mtcars %>% mean_by(by = cyl, var = disp)
```

Learn more about this pattern in [Data mask programming patterns](https://rlang.r-lib.org/reference/topic-data-mask-programming.html).

---

Reply:
Should `rlang` be added to the DESCRIPTION?

Reply:
We figured out why we need the `{{}}`, it's because `readr::read_delim()` uses` tidyverse dplyr::select()` like syntax, so any symbol passed to the function will be interpreted as a column name, in order to parse the symbol we need to initiate data-masking by embracing the symbol. 

A comment needs to be added to make this clear.